core/src/wallet/algorand/database/AlgorandTransactionDatabaseHelper.hpp

```diff
@@ -32,13 +32,122 @@
 
 #include "../model/transactions/AlgorandTransaction.hpp"
 
+#include <math/BaseConverter.hpp>
 #include <boost-optional.h>
 #include <database/soci-number.h>
 
 namespace ledger {
 namespace core {
 namespace algorand {
 
+    static constexpr auto COL_TX_HASH = 1;
+    static constexpr auto COL_TX_TYPE = 2;
+    static constexpr auto COL_TX_ROUND = 3;
+    static constexpr auto COL_TX_TIMESTAMP = 4;
+    static constexpr auto COL_TX_FIRST_VALID = 5;
+    static constexpr auto COL_TX_LAST_VALID = 6;
+    static constexpr auto COL_TX_GENESIS_ID = 7;
+    static constexpr auto COL_TX_GENESIS_HASH = 8;
+    static constexpr auto COL_TX_SENDER = 9;
+    static constexpr auto COL_TX_FEE = 10;
+    static constexpr auto COL_TX_NOTE = 11;
+    static constexpr auto COL_TX_GROUP = 12;
+    static constexpr auto COL_TX_LEASE = 13;
+    static constexpr auto COL_TX_SENDER_REWARDS = 14;
+    static constexpr auto COL_TX_RECEIVER_REWARDS = 15;
+    static constexpr auto COL_TX_CLOSE_REWARDS = 16;
+
+    static constexpr auto COL_TX_PAY_AMOUNT = 17;
+    static constexpr auto COL_TX_PAY_RECEIVER_ADDRESS = 18;
+    static constexpr auto COL_TX_PAY_CLOSE_ADDRESS = 19;
+    static constexpr auto COL_TX_PAY_CLOSE_AMOUNT = 20;
+
+    static constexpr auto COL_TX_KEYREG_NONPART = 21;
+    static constexpr auto COL_TX_KEYREG_SELECTION_PK = 22;
+    static constexpr auto COL_TX_KEYREG_VOTE_PK = 23;
+    static constexpr auto COL_TX_KEYREG_VOTE_KEY_DILUTION = 24;
+    static constexpr auto COL_TX_KEYREG_VOTE_FIRST = 25;
+    static constexpr auto COL_TX_KEYREG_VOTE_LAST = 26;
+
+    static constexpr auto COL_TX_ACFG_ASSET_ID = 27;
+    static constexpr auto COL_TX_ACFG_ASSET_NAME = 28;
+    static constexpr auto COL_TX_ACFG_UNIT_NAME = 29;
+    static constexpr auto COL_TX_ACFG_TOTAL = 30;
+    static constexpr auto COL_TX_ACFG_DECIMALS = 31;
+    static constexpr auto COL_TX_ACFG_DEFAULT_FROZEN = 32;
+    static constexpr auto COL_TX_ACFG_CREATOR_ADDRESS = 33;
+    static constexpr auto COL_TX_ACFG_MANAGER_ADDRESS = 34;
+    static constexpr auto COL_TX_ACFG_RESERVE_ADDRESS = 35;
+    static constexpr auto COL_TX_ACFG_FREEZE_ADDRESS = 36;
+    static constexpr auto COL_TX_ACFG_CLAWBACK_ADDRESS = 37;
+    static constexpr auto COL_TX_ACFG_METADATA_HASH = 38;
+    static constexpr auto COL_TX_ACFG_URL = 39;
+
+    static constexpr auto COL_TX_AXFER_ASSET_ID = 40;
+    static constexpr auto COL_TX_AXFER_ASSET_AMOUNT = 41;
+    static constexpr auto COL_TX_AXFER_RECEIVER_ADDRESS = 42;
+    static constexpr auto COL_TX_AXFER_CLOSE_ADDRESS = 43;
+    static constexpr auto COL_TX_AXFER_CLOSE_AMOUNT = 44;
+    static constexpr auto COL_TX_AXFER_SENDER_ADDRESS = 45;
+
+    static constexpr auto COL_TX_AFRZ_ASSET_ID = 46;
+    static constexpr auto COL_TX_AFRZ_FROZEN = 47;
+    static constexpr auto COL_TX_AFRZ_FROZEN_ADDRESS = 48;
+
+
+    // Helpers to deal with Option<> types,
+    // because SOCI only understands boost::optional<>...
+
```

Comment:
a question: why do you move a lot of methods from cpp file to hpp file?
because I think this hpp is included by other files and it will increase the compilation time in theory.

Reply:
because those functions are called in AlgorandOperationsDatabaseHelper.cpp